sys/sys/malloc.h

```diff
@@ -232,7 +232,7 @@ void	*malloc(size_t size, struct malloc_type *type, int flags) __malloc_like
  */
 #define	malloc(size, type, flags) ({					\
 	void *_malloc_item;						\
-	size_t _size = (size);						\
+	size_t _size = CHERI_REPRESENTABLE_LENGTH(size);		\
```

Comment:
One suggestion: do we want to signpost this in the code via a comment that we should axe this in the future when/if the compiler changes?